build.gradle

```diff
@@ -67,48 +71,40 @@ sourceSets {
     }
 }
 
-pluginBundle {
-    website = 'https://github.com/gradle/android-cache-fix-gradle-plugin'
-    vcsUrl = 'https://github.com/gradle/android-cache-fix-gradle-plugin'
-
-    description = 'Gradle plugin to fix Android caching problems'
-
+gradlePlugin {
     plugins {
         androidCacheFixPlugin {
             id = 'org.gradle.android.cache-fix'
             displayName = 'Gradle Android cache fix plugin'
-            tags = ['android', 'cache', 'fix']
-            version = project.version
+            description = 'Gradle plugin to fix Android caching problems'
+            implementationClass = 'org.gradle.android.AndroidCacheFixPlugin'
         }
     }
 }
 
-def localRepo = file("$buildDir/local-repo")
+pluginBundle {
+    website = 'https://github.com/gradle/android-cache-fix-gradle-plugin'
+    vcsUrl = 'https://github.com/gradle/android-cache-fix-gradle-plugin'
 
-artifacts {
-    runtime jar
+    tags = ['android', 'cache', 'fix']
 }
 
-tasks.register('install', Upload) {
+def localRepo = file("$buildDir/local-repo")
+
+publishing {
     repositories {
-        ivy {
+        maven {
             url = localRepo.toURI()
         }
     }
-    descriptorDestination = file("$buildDir/ivy/ivy.xml")
-    uploadDescriptor = true
-    configuration = configurations.runtime
 }
 
 tasks.withType(Test).configureEach {
-    dependsOn install
+    dependsOn publish
     systemProperty "local.repo", localRepo.toURI()
+    maxParallelForks = 4
```

Comment:
Did you find that this actually improved anything?  When I've tried this before, with the android builds these tests execute, I found that it didn't improve the execution time at all, but it made my system nearly unusable while the tests were running.  It appeared that it was almost completely I/O bound, so adding more parallelism didn't buy us anything.

Reply:
I can't say I really noticed, but tests seemed super-slow without it. 
Looking at the CI job, it's probably not any faster. I'll remove this.

Reply:
Haha, yeah I did this too at some point :)